catalog/spatial/geocoding/spatial-geocoding-extract/src/main/java/org/codice/ddf/spatial/geocoding/extract/GeoNamesFileExtractor.java

```diff
@@ -144,15 +148,20 @@ public void pushGeoEntriesToExtractionCallback(
   private InputStream getInputStreamFromResource(
       String resource, ProgressCallback extractionCallback)
       throws GeoEntryExtractionException, GeoNamesRemoteDownloadException {
-
-    if (FilenameUtils.isExtension(resource, "zip")) {
+    if (!urlValidator.isValid(resource) && FilenameUtils.isExtension(resource, "zip")) {
```

Comment:
❓ Why do we need to check urlValidator.isValid(resource) in this method? Can you extract it to a variable so that you don't need to check it multiple times?

Reply:
❓ Are file paths URLs? What if someone entered file://...? Would this be handled by `webClient = WebClient.create(resource);`?

Reply:
🤷‍♀️ Maybe it'd be worth adding a new unit test for this case.